code/modules/ai/action_states/action_state.dm

```diff
@@ -0,0 +1,199 @@
+
+/datum/action_state
+	var/datum/component/ai_behavior/parent_ai
+
+/datum/action_state/New(parent_to_hook_to)
+	..()
+	if(parent_to_hook_to)
+		parent_ai = parent_to_hook_to
+		return TRUE
+	else
+		stack_trace("A action state was initialized without a parent to hook onto, VERY BAD!!!")
+		qdel(src)
+		return FALSE
+
+//Called from AI process, this is NOT for assigning new states or removing and choosing next state, that's the AI behavior job
+//This is for updating things like random_move's next_node
+
+/datum/action_state/proc/Process()
```

Comment:
datums have process()

Reply:
Renamed Process() to process() and did START_PROCESSING() and STOP_PROCESSING() on new() and Destroy() respectively